contrib/curl-cmake/CMakeLists.txt

```diff
@@ -1,152 +1,187 @@
+            ${LIBRARY_DIR}/lib/altsvc.c
+            ${LIBRARY_DIR}/lib/socketpair.c
+            ${LIBRARY_DIR}/lib/vauth/vauth.c
+            ${LIBRARY_DIR}/lib/vauth/cleartext.c
+            ${LIBRARY_DIR}/lib/vauth/cram.c
+            ${LIBRARY_DIR}/lib/vauth/digest.c
+            ${LIBRARY_DIR}/lib/vauth/digest_sspi.c
+            ${LIBRARY_DIR}/lib/vauth/krb5_gssapi.c
+            ${LIBRARY_DIR}/lib/vauth/krb5_sspi.c
+            ${LIBRARY_DIR}/lib/vauth/ntlm.c
+            ${LIBRARY_DIR}/lib/vauth/ntlm_sspi.c
+            ${LIBRARY_DIR}/lib/vauth/oauth2.c
+            ${LIBRARY_DIR}/lib/vauth/spnego_gssapi.c
+            ${LIBRARY_DIR}/lib/vauth/spnego_sspi.c
+            ${LIBRARY_DIR}/lib/vtls/openssl.c
+            ${LIBRARY_DIR}/lib/vtls/gtls.c
+            ${LIBRARY_DIR}/lib/vtls/vtls.c
+            ${LIBRARY_DIR}/lib/vtls/nss.c
+            ${LIBRARY_DIR}/lib/vtls/polarssl.c
+            ${LIBRARY_DIR}/lib/vtls/polarssl_threadlock.c
+            ${LIBRARY_DIR}/lib/vtls/wolfssl.c
+            ${LIBRARY_DIR}/lib/vtls/schannel.c
+            ${LIBRARY_DIR}/lib/vtls/schannel_verify.c
+            ${LIBRARY_DIR}/lib/vtls/sectransp.c
+            ${LIBRARY_DIR}/lib/vtls/gskit.c
+            ${LIBRARY_DIR}/lib/vtls/mbedtls.c
+            ${LIBRARY_DIR}/lib/vtls/mesalink.c
+            ${LIBRARY_DIR}/lib/vtls/bearssl.c
+            ${LIBRARY_DIR}/lib/vquic/ngtcp2.c
+            ${LIBRARY_DIR}/lib/vquic/quiche.c
+            ${LIBRARY_DIR}/lib/vssh/libssh2.c
+            ${LIBRARY_DIR}/lib/vssh/libssh.c
+        )
 
-target_compile_definitions(curl PRIVATE OS="${CMAKE_SYSTEM_NAME}")
+        add_library (curl ${SRCS})
 
-target_link_libraries(curl PRIVATE ssl)
+        target_compile_definitions (curl PRIVATE
+            HAVE_CONFIG_H
+            BUILDING_LIBCURL
+            CURL_HIDDEN_SYMBOLS
+            libcurl_EXPORTS
+            OS="${CMAKE_SYSTEM_NAME}"
+        )
+        target_include_directories (curl PUBLIC
+            ${LIBRARY_DIR}/include
+            ${LIBRARY_DIR}/lib
+            . # curl_config.h
+        )
+
+        target_link_libraries (curl PRIVATE ssl)
+
+        # The library is large - avoid bloat (XXX: is it?)
+        target_compile_options (curl PRIVATE -g0)
+
+        # find_package(CURL) compatibility for the following packages that uses
+        # find_package(CURL)/include(FindCURL):
+        # - mariadb-connector-c
+        # - aws-s3-cmake
+        # - sentry-native
+        set (CURL_FOUND ON CACHE BOOL "")
+        set (CURL_ROOT_DIR ${LIBRARY_DIR} CACHE PATH "")
+        set (CURL_INCLUDE_DIR ${LIBRARY_DIR}/include CACHE PATH "")
+        set (CURL_INCLUDE_DIRS ${LIBRARY_DIR}/include CACHE PATH "")
+        set (CURL_LIBRARY curl CACHE STRING "")
+        set (CURL_LIBRARIES ${CURL_LIBRARY} CACHE STRING "")
+        set (CURL_VERSION_STRING 7.67.0 CACHE STRING "")
+        add_library (CURL::libcurl ALIAS ${CURL_LIBRARY})
+    else ()
+        find_package (CURL)
+    endif ()
+endif ()
```

Comment:
We don't have a build for `ENABLE_LIBRARIES=0` for now, but it will be good to also handle the case when `ENABLE_CURL=0`. If it shouldn't work, then there is no need for this option in the first place.

The point is to link with `curl` always, even if the library is disabled. In this case there should be just an empty `INTERFACE` library (maybe with public defined macro `USE_CURL=0`).

Reply:
>We don't have a build for ENABLE_LIBRARIES=0 for now, but it will be good to also handle the case when ENABLE_CURL=0. If it shouldn't work, then there is no need for this option in the first place.

Are you suggesting to verify was CURL enabled in it's users (sentry, s3, mysql connector) and bail/disable if not?

>The point is to link with curl always, even if the library is disabled. In this case there should be just an empty INTERFACE library (maybe with public defined macro USE_CURL=0).

What are the benefits of providing empty library in this case?
There are 3 libraries that will try finding CURL anyway.

Reply:
If ENABLE_CURL will be dropped then in case of unbundled build, curl library will always be searched fail if there is no system-wide curl (will add `REQUIRED` to avoid `if`ing each user, since this also looks like "heuristic")